scripts/sign.py

```diff
@@ -4,55 +4,99 @@
+
+    '   command:\n' +
+    '     sign             Generate signed loadable TA image file.\n' +
+    '                      Takes arguments --uuid, --in, --out and --key.\n' +
+    '     generate-digest  Generate loadable TA binary image digest for offline\n' +
+    '                      signing. Takes arguments  --uuid, --in and --out.\n' +
+    '     stitch-ta        Generate loadable signed TA binary image file from\n' +
+    '                      TA raw image and its signature. Takes arguments\n' +
+    '                      --uuid, --in, --out, and --sig.\n\n' +
+
+    'command_opts\n' +
+    '    --version NUM    NUM shall be 0. Defaults to 0.\n\n' +
+
+    'command_arg\n' +
+    '    --uuid UUID      String UUID of the TA\n' +
+    '    --in INFILE      Input raw unsigned TA image file path\n' +
+    '    --out OUTFILE    Generated output file path\n' +
+    '    --key PRIVKEY    Private key in PEM format\n' +
+    '    --sig SIGNATURE  TA image signature binary file\n' +
+    '    --dig DIGEST     TA image digest binary file\n',
+               epilog='If no command is given, the script will default to "sign".')
+
+    # when choices= is used, the command can no longer be left out and the default used
+    #parser.add_argument('command', choices=command_choices, nargs='?', default='sign', help='Command, one of [' + ', '.join(command_choices) + ']')
+    parser.add_argument('command', nargs='?', default='sign', help='Command, one of [' + ', '.join(command_choices) + ']')
     parser.add_argument('--uuid', required=True,
-                        type=uuid_parse, help='UUID of TA')
+                        type=uuid_parse, help='String UUID of TA')
     parser.add_argument('--version', type=int_parse, default=0, help='Version')
     parser.add_argument('--key', required=True, help='Name of key file')
-    parser.add_argument('--in', required=True, dest='inf',
-                        help='Name of in file')
-    parser.add_argument('--out', required=True, help='Name of out file')
-    return parser.parse_args()
-
+    parser.add_argument('--sig', required=False, dest='sigf', help='Name of signature input file, defaults to <UUID>.sig')
+    parser.add_argument('--dig', required=False, dest='digf', help='Name of digest output file, defaults to <UUID>.dig')
+    parser.add_argument('--in', required=False, dest='inf',
+                        help='Name of application in file, defaults to <UUID>.stripped.elf')
+    parser.add_argument('--out', required=False, dest='outf', help='Name of application out file, defaults to <UUID>.ta')
+
+    # set defauts
+
+    parsed = parser.parse_args()
+    if parsed.sigf is None:
+        parsed.sigf = str(parsed.uuid)+'.sig'
+    if parsed.digf is None:
+        parsed.digf = str(parsed.uuid)+'.dig'
+    if parsed.inf is None:
+        parsed.inf = str(parsed.uuid)+'.stripped.elf'
+    if parsed.outf is None:
+        parsed.outf = str(parsed.uuid)+'.ta'
+
+    return parsed
```

Comment:
add an empty line below.
Check your python script against `pep8` or like tool; it will well you (@jforissier is a ref).

Reply:
Yes, next PR update should be `pep8` clean.